src/bun.js/api/bun/ms.zig

```diff
@@ -0,0 +1,308 @@
+/// Parse a time string like "2d", "1.5h", "5m" to milliseconds
+pub fn parse(input: []const u8) ?f64 {
+    if (input.len == 0 or input.len > 100) return null;
+
+    var i: usize = 0;
+
+    next: switch (input[i]) {
+        '-', '.', '0'...'9' => {
+            i += 1;
+            if (i < input.len) continue :next input[i];
+            break :next;
+        },
+        ' ', 'a'...'z', 'A'...'Z' => break :next,
+        else => return null,
+    }
+
+    const value = std.fmt.parseFloat(f64, input[0..i]) catch return null;
+
+    const unit = strings.trimLeadingChar(input[i..], ' ');
+    if (unit.len == 0) return value;
+
+    if (MultiplierMap.getASCIIICaseInsensitive(unit)) |m| {
+        return value * m;
+    }
+
+    return null;
+}
+
+// Years (365.25 days to account for leap years)
+// (matching the `ms` package implementation)
+const ms_per_year = std.time.ms_per_day * 365.25;
+const ms_per_month = std.time.ms_per_day * (365.25 / 12.0);
+
+const MultiplierMap = bun.ComptimeStringMap(f64, .{
+    // Years (365.25 days to account for leap years)
+    .{ "y", ms_per_year },
+    .{ "yr", ms_per_year },
+    .{ "yrs", ms_per_year },
+    .{ "year", ms_per_year },
+    .{ "years", ms_per_year },
+
+    // Months (30.4375 days average)
+    .{ "mo", ms_per_month },
+    .{ "month", ms_per_month },
+    .{ "months", ms_per_month },
+
+    // Weeks
+    .{ "w", std.time.ms_per_week },
+    .{ "week", std.time.ms_per_week },
+    .{ "weeks", std.time.ms_per_week },
+
+    // Days
+    .{ "d", std.time.ms_per_day },
+    .{ "day", std.time.ms_per_day },
+    .{ "days", std.time.ms_per_day },
+
+    // Hours
+    .{ "h", std.time.ms_per_hour },
+    .{ "hr", std.time.ms_per_hour },
+    .{ "hrs", std.time.ms_per_hour },
+    .{ "hour", std.time.ms_per_hour },
+    .{ "hours", std.time.ms_per_hour },
+
+    // Minutes
+    .{ "m", std.time.ms_per_min },
+    .{ "min", std.time.ms_per_min },
+    .{ "mins", std.time.ms_per_min },
+    .{ "minute", std.time.ms_per_min },
+    .{ "minutes", std.time.ms_per_min },
+
+    // Seconds
+    .{ "s", std.time.ms_per_s },
+    .{ "sec", std.time.ms_per_s },
+    .{ "secs", std.time.ms_per_s },
+    .{ "second", std.time.ms_per_s },
+    .{ "seconds", std.time.ms_per_s },
+
+    // Milliseconds
+    .{ "ms", 1 },
+    .{ "msec", 1 },
+    .{ "msecs", 1 },
+    .{ "millisecond", 1 },
+    .{ "milliseconds", 1 },
+});
+
+// To keep the behavior consistent with JavaScript, we can't use @round
+// Zig's @round uses "round half away from zero": ties round away from zero (2.5→3, -2.5→-3)
+// JavaScript's Math.round uses "round half toward +∞": ties round toward positive infinity (2.5→3, -2.5→-2)
+fn jsMathRound(x: f64) i64 {
+    const i: f64 = @ceil(x);
+    if ((i - 0.5) > x) return @intFromFloat(i - 1.0);
+    return @intFromFloat(i);
+}
+
+/// Format milliseconds to a human-readable string
+pub fn format(allocator: std.mem.Allocator, ms: f64, long: bool) ![]u8 {
```

Comment:
I think you could do a lot better in this function. I will offer you a proposal PR, since expressing my ideas in comment form would be inefficient. Will reply to this comment when I have a PR.